infrastructure/zk/src/lint.ts

```diff
@@ -36,14 +36,19 @@ async function lintSystemContracts(check: boolean = false) {
     await utils.spawn(`yarn --silent --cwd etc/system-contracts lint:${check ? 'check' : 'fix'}`);
 }
 
+function clippyRustToolchain(): string {
+    //There are no specific reasons to run this particular version other than because clippy is not very stable
+    // and even the same clippy sem-ver version can result in different output based on compiler version
+    return 'nightly-2023-08-21';
+}
```

Comment:
Fixing this here may break a lot of stuff because our nightly builds are going to be built with the actual nightly compilers each new day (or at least that is the idea). I would advice just not to override rust in your builds and default to what the repo specifies.